pkg/cluster/graph/manager.go

```diff
@@ -121,3 +127,43 @@ func (m *manager) LoadPersisted(ctx context.Context, resourceGroup, account stri
 
 // SavePersistedGraph could be implemented and used with care if needed, but
 // currently we don't need it (and it's better that way)
+
+// GetUserDelegatedSASIgnitionBlobURL is used for MIWI clusters so that Ignition blob can be accessed by bootstrap VM without Storage Account Shared Access Keys
+func (m *manager) GetUserDelegatedSASIgnitionBlobURL(ctx context.Context, resourceGroup, account, blobURL string, usesWorkloadIdentity bool) (string, error) {
+	if !usesWorkloadIdentity {
+		return "", fmt.Errorf("getUserDelegatedSASIgnitionBlobURL called for a Cluster Service Principal cluster")
+	}
+	urlParts, err := sas.ParseURL(blobURL)
+	if err != nil {
+		return "", err
+	}
+	currentTime := time.Now().UTC().Add(-10 * time.Second)
+	expiryTime := time.Now().UTC().Add(time.Hour)
+	perms := sas.BlobPermissions{Read: true}
```

Comment:
I noticed that for CSP clusters we give "read" and "list", so why does only "read" work here?

Reply:
Good point, I didn't notice that other SAS token certificate requests the list permission, and I am not sure why it is required for AccountSAS.
Here is the reference that was used to determine the permissions:-
User Delegated : https://learn.microsoft.com/en-us/rest/api/storageservices/create-user-delegation-sas#specify-permissions

For AccountSAS: https://learn.microsoft.com/en-us/rest/api/storageservices/create-account-sas#specify-the-account-sas-parameters

Since we define the direct URL for the blob access, I don't think `List` is needed anywhere. But I also don't want to change the pre-existing implementation for the same.
User Delegated flow is being tested with Read Only so it works.

Reply:
>Since we define the direct URL for the blob access, I don't think List is needed anywhere.

Makes sense, thanks for explaining!